src/map/skill.cpp

```diff
@@ -12959,22 +12949,16 @@ int32 skill_castend_nodamage_id (struct block_list *src, struct block_list *bl,
 		}
 		break;
 	case RL_QD_SHOT:
-		if (sd) {
-			skill_area_temp[1] = bl->id;
-			// Check surrounding
-			skill_area_temp[0] = map_foreachinrange(skill_area_sub, src, skill_get_splash(skill_id, skill_lv), BL_CHAR, src, skill_id, skill_lv, tick, BCT_ENEMY, skill_area_sub_count);
-			if (skill_area_temp[0])
-				map_foreachinallrange(skill_area_sub, src, skill_get_splash(skill_id, skill_lv), BL_CHAR, src, skill_id, skill_lv, tick, flag|BCT_ENEMY|SD_SPLASH|1, skill_castend_damage_id);
+		// Check surrounding area and hit all targets marked with crimson marker except main target
+		skill_area_temp[1] = bl->id;
+		map_foreachinallrange(skill_area_sub, src, skill_get_splash(skill_id, skill_lv), BL_CHAR, src, skill_id, skill_lv, tick, flag|BCT_ENEMY|SD_SPLASH|1, skill_castend_damage_id);
 
-			// Main target always receives damage
-			clif_skill_nodamage(src, *src, skill_id, skill_lv);
-			skill_attack(skill_get_type(skill_id), src, src, bl, skill_id, skill_lv, tick, flag|BCT_ENEMY|SD_LEVEL);
-		} else {
-			clif_skill_nodamage(src, *src, skill_id, skill_lv);
-			map_foreachinrange(skill_area_sub, src, skill_get_splash(skill_id, skill_lv), BL_CHAR, src, skill_id, skill_lv, tick, flag|BCT_ENEMY|SD_SPLASH|1, skill_castend_damage_id);
-		}
-		status_change_end(src, SC_QD_SHOT_READY); // End here to prevent spamming of the skill onto the target.
-		skill_area_temp[0] = 0;
+		// Main target always receives damage
+		clif_skill_nodamage(src, *src, skill_id, skill_lv);
+		skill_attack(skill_get_type(skill_id), src, src, bl, skill_id, skill_lv, tick, flag|BCT_ENEMY|SD_LEVEL);
```

Comment:
Should be moved over to skill_castend_damage_id.
There we check: `if (skill_area_temp[1] == bl->id)` already and it could be turned into a simple if/else logic there.

Reply:
I was thinking that too, but there was a deviation in the code. One is using SD_LEVEL and the other is using SD_SPLASH and I wanted to avoid functional changes.

Trying to get a packet capture for this at the moment, but didn't plan to include it in this PR. 
(Usually I would just fix the whole skill, but for the sake of small PRs I only wanted to do a cleanup here.)

Reply:
Upon further investigation it looks like it sends the same damage packet to all targets including the current one, so I guess you're right and I can optimize it further.

Reply:
How about now?